### PR TITLE
code - Sum of digits of a number

### PR DESCRIPTION
**Pull request by sarayusreeyadavpadala**
@geekcomputers I have added the changes to my previous code.

### CODE REVIEW

File: Sum of digits of a number.py [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
-   r=n%10           # Here, we are extracting each digit from "n" starting from one's place to ten's and hundred's... so on.
+def get_integer():
+    for i in range(3,0,-1):                       # executes the loop 3 times. Giving 3 chances to the user.
+        num = input("enter a number:")
+        if num.isnumeric():                       # checks if entered input is an integer string or not.
+            num = int(num)                        # converting integer string to integer. And returns it to where function is called.
+            return num
+        else:
+            print("enter integer only")                    
+            if (i-1)<=1:                                   # if user entered wrong input, it displays a message and chances left.
```
Comment (NitkarshChourasia) [inline Sum of digits of a number.py:13]: no need to use if and else.
You can use ternary operator.
Like this:

print(f"{i-1} {'chance' if i-1 == 1 else 'chances'} left")
This, does the trick.

File: Sum of digits of a number.py [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+        continue   
+                
+def addition(num):
+    Sum=0
+    if type(num) is type(None):               # Checks if number type is none or not. If type is none program exits.
+        print("Try again!")
+        sys.exit()
+    while num > 0:                            # Addition- adding the digits in the number.
+        digit = int(num % 10)
+        Sum += digit
```
Comment (NitkarshChourasia) [inline Sum of digits of a number.py:27]: Also use doc-string on the main function.
To explain it when.
print(function.\_\_doc\_\_  ) is used.

**Review by NitkarshChourasia — CHANGES_REQUESTED** [review]

File: Sum of digits of a number.py (hunk 2 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:27]: why this? @NitkarshChourasia

Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:27]: > Also use doc-string on the main function. To explain it when. print(function.__doc__ ) is used.

why this?

File: Sum of digits of a number.py (hunk 1 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py]: @NitkarshChourasia I have added this. review my code

**Comment by sarayusreeyadavpadala** [comment]
I have changed my code. Help me to improve this further. Needs your suggestion.....

**Comment by NitkarshChourasia** [comment]
tomorrow morning I will review.
Hey, listen I have a project coming up.
Would you mind, become a code reviewer on it? 
It will be based on python.
but what other languages do you know?
I will be doing...
c++, javascript, and c languages.
and in python...
machine learning, statistics, and what not.

Are you ready to take part in all of these?

**Comment by sarayusreeyadavpadala** [comment]
> @sarayusreeyadavpadala
> 
> Okay?? @sarayusreeyadavpadala

Okay @NitkarshChourasia

**Comment by sarayusreeyadavpadala** [comment]
> @sarayusreeyadavpadala Always tag me like this... @NitkarshChourasia to let me know, that you are asking specifically to me only, and not somebody else.

Yeah ok @NitkarshChourasia

**Comment by sarayusreeyadavpadala** [comment]
> tomorrow morning I will review. Hey, listen I have a project coming up. Would you mind, become a code reviewer on it? It will be based on python. but what other languages do you know? I will be doing... c++, javascript, and c languages. and in python... machine learning, statistics, and what not.
> 
> Are you ready to take part in all of these?

Thanks @NitkarshChourasia 
Yeah definitely, I'll review your code.
I know C++, C, Python, a little javascript.
But I don't have in depth knowledge in it.

Yeah! I am ready to take a part in this.
@NitkarshChourasia

**Comment by sarayusreeyadavpadala** [comment]
> It will be basic programs, you just have to review them and provide feedback, to whatever extent you can. If you have any colleague you can tell them to join, too. @sarayusreeyadavpadala

Okay! I'll do that. @NitkarshChourasia

**Comment by NitkarshChourasia** [comment]
> > tomorrow morning I will review. Hey, listen I have a project coming up. Would you mind, become a code reviewer on it? It will be based on python. but what other languages do you know? I will be doing... c++, javascript, and c languages. and in python... machine learning, statistics, and what not.
> > Are you ready to take part in all of these?
> 
> Thanks @NitkarshChourasia Yeah definitely, I'll review your code. I know C++, C, Python, a little javascript. But I don't have in depth knowledge in it.
> 
> Yeah! I am ready to take a part in this. @NitkarshChourasia

Yeah, it will be a learning experience, Indeed.
I am starting and trying myself from base printing function zero....to ....hero...suppose.

**Comment by NitkarshChourasia** [comment]
@sarayusreeyadavpadala ASK ALL YOUR FRIENDS TO PARTICIPATE FOR IT.
I WILL BE NEEDING THEIR USERNAME TO ADD TO THE REPOSITORY AS A REVIEWER, I GUESS.

File: Sum of digits of a number.py [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
-   r=n%10           # Here, we are extracting each digit from "n" starting from one's place to ten's and hundred's... so on.
+def get_integer():
+    for i in range(3,0,-1):                       # executes the loop 3 times. Giving 3 chances to the user.
+        num = input("enter a number:")
+        if num.isnumeric():                       # checks if entered input is an integer string or not.
+            num = int(num)                        # converting integer string to integer. And returns it to where function is called.
+            return num
+        else:
+            print("enter integer only")                    
+            print(f'{i-1} chances are left' if (i-1)>1 else f'{i-1} chance is left')     # prints if user entered wrong input and chances left.
```
Comment (NitkarshChourasia) [inline Sum of digits of a number.py:13]: print(f"{i-1} {'chance' if i-1 == 1 else 'chances'} left")

This is better readability I think.
But, your is fine, too!

**Review by NitkarshChourasia — APPROVED** [review]

File: Sum of digits of a number.py (hunk 3 above)
Comment (sarayusreeyadavpadala) [reply Sum of digits of a number.py:13]: Thank you @NitkarshChourasia

**Comment by NitkarshChourasia** [comment]
@sarayusreeyadavpadala Congratulations.

**Comment by NitkarshChourasia** [comment]
@sarayusreeyadavpadala Welcome.

**Pull request merged**